ConwaysGameOfLife/Universe.cs

```diff
@@ -67,23 +65,25 @@ public List<Location> GetCellNeighbourLocations(Cell cell)
             return neighbourLocations;
         }
 
-        public CellState GetCellStateFromLocation(Location location)
+        public CellState GetCellStateFromLocation(Location location) //TODO: do I need this - can use get cellaatlocation
         {
-            var newCells = Cells.Where(cell => cell.Location == new Location(cell.Location.X, cell.Location.Y));
-            var newCell = newCells.ElementAt(0);
-            return newCell.CellState;
+            // var newCell = Cells.Find(cell => cell.Location == new Location(cell.Location.X, cell.Location.Y));
+            var x = location.X;
+            var y = location.Y;
+            var cellAtLocation = Cells.FirstOrDefault(cell => cell.Location.X == x && cell.Location.Y == y);
+            return cellAtLocation.CellState;
         }
 
         public List<Cell> CheckIfNeighboursAlive(Cell cell) 
         {
             var neighbourList = new List<Cell>();
-            _neighbourLocations = GetCellNeighbourLocations(cell);
-            foreach (var neighbour in _neighbourLocations)
+            var neighbourLocations = GetCellNeighbourLocations(cell);
+            foreach (var neighbour in neighbourLocations)

```

Comment:
Could potentially use Linq:

```suggestion
            aliveNeighbours = neighbourLocations.Count(location => GetCellStateFromLocation(location) == CellState.Alive);
```